parity/account_utils.rs

```diff
@@ -145,22 +147,25 @@ mod accounts {
 			return Err(format!("No password found for the consensus signer {}. {}", engine_signer, VERIFY_PASSWORD_HINT));
 		}
 
-		let mut author = None;
-		for password in passwords {
+		let mut invalid_reasons = std::collections::HashSet::new();
+		for (idx, password) in passwords.iter().enumerate() {
 			let signer = parity_rpc::signer::EngineSigner::new(
 				account_provider.clone(),
 				engine_signer,
 				password.clone(),
 			);
-			if signer.sign(Default::default()).is_ok() {
-				author = Some(::ethcore::miner::Author::Sealer(Box::new(signer)));
+			if let Err(e) = signer.sign(SECP_TEST_MESSAGE) {
+				debug!(target: "account", "Signing test of `EngineSigner ({})` with password index: {} failed because of: {:?}", engine_signer, idx, e);
+				invalid_reasons.insert(e.to_string());
+			} else {
+				return Ok(Some(ethcore::miner::Author::Sealer(Box::new(signer))));
 			}
 		}
-		if author.is_none() {
-			return Err(format!("No valid password for the consensus signer {}. {}", engine_signer, VERIFY_PASSWORD_HINT));
-		}
 
-		Ok(author)
+		Err(format!(
```

Comment:
This will look like:

```bash
No valid password found for EngineSigner 0x5100…cae4, the following errors were found during testing: {"custom crypto error: invalid AES message", "custom crypto error: Invalid password"}. Make sure valid password is present in files passed using `--password` or in the configuration file.
```